Description: Allow miri to be built on the stable channel
Author: Max Gilmour <max.gilmour@canonical.com>
Forwarded: not-needed
Last-Update: 2026-01-20
---
Index: rustc-1.97.0-src/src/bootstrap/src/core/build_steps/dist.rs
===================================================================
--- rustc-1.97.0-src.orig/src/bootstrap/src/core/build_steps/dist.rs
+++ rustc-1.97.0-src/src/bootstrap/src/core/build_steps/dist.rs
@@ -1582,9 +1582,10 @@ impl Step for Miri {
         // This prevents miri from being built for "dist" or "install"
         // on the stable/beta channels. It is a nightly-only tool and should
         // not be included.
-        if !builder.build.unstable_features() {
-            return None;
-        }
+        // UBUNTU: Allow building miri on stable channel
+        // if !builder.build.unstable_features() {
+        //     return None;
+        // }
 
         let miri = builder.ensure(tool::Miri::from_compilers(self.compilers));
         let cargomiri = builder.ensure(tool::CargoMiri::from_compilers(self.compilers));
